src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCache.scala

```diff
@@ -97,35 +77,13 @@ class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference) extends Acto
 
   private def hostToIpEntry(c: Cluster): (Host, IP) = host(c) -> c.dataprocInfo.hostIp.get
 
-  private def projectNameToHostEntry(c: Cluster): ((GoogleProject, ClusterName), GetClusterResponse) = {
+  private def projectNameToHostEntry(c: Cluster): ((GoogleProject, ClusterName), HostStatus) = {
```

Comment:
We can use the newly created `DnsCacheKey` instead of `(GoogleProject, ClusterName)` so

```scala
  private def projectNameToHostEntry(c: Cluster): (DnsCacheKey, HostStatus) = ???
```

Reply:
Actually, `projectNameToHostEntry` is called from only within `projectNameToHost.load()` where only the second element (`._2`) of the output tuple is used. Therefore, I'd suggest changing the method as:

```scala
private def clusterToHostStatus(c: Cluster): HostStatus = {
  if (c.status.isStartable)
    HostPaused
  else if (c.dataprocInfo.hostIp.isDefined)
    HostReady(host(c))
  else
    HostNotReady
}
```